astropy/units/tests/test_quantity.py

```diff
@@ -1003,6 +1003,169 @@ def test_to_string(self):
         res = "[0 1 2] (Unit not initialised)"
         assert np.arange(3).view(u.Quantity).to_string() == res
 
+    @pytest.mark.parametrize(
+        "input_value, input_unit, format_spec, expected_result, format, precision",
+        [
+            (1.5e14, "m/s", ".2e", "1.50e+14 m / s", None, None),
+            (0.123, "m/s", "0.3f", "0.123 m / s", None, None),
+            (0.000123, "m/s", ".2e", "1.23e-04 m / s", None, None),
+            (1.23456789e15, "m/s", ".2e", "1.23e+15 m / s", None, 5),
```

Comment:
This is the only parametrization that has a `precision` different from `None`, which is suggesting that `precision` should not be a parameter for this test and it should be tested separately instead.